src/dynamics_tendencies.jl

```diff
@@ -0,0 +1,216 @@
+"""
+Compute the spectral tendency of the surface pressure logarithm
+"""
+function surface_pressure_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                   Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   l2::Int,                       # leapfrog index 2 (time step used for tendencies)
+                                   C::Constants{NF}
+                                   ) where {NF<:AbstractFloat}
+
+
+    @unpack pres_surf= Prog 
+    @unpack pres_surf_tend = Diag.Tendencies
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,dumc,px,py = Diag.miscvars
+    @unpack nlev, dhs = C 
+
+    #Initialise mean fields. 
+    u_mean[:,:] = 0.0
+    v_mean[:,:] = 0.0
+    d_mean[:,:] = 0.0 #d-mean is calculated here but actually used in vertical_velocity_tendency!()
+
+    #..and calculate values
+    for k in 1:nlev
+        u_mean += u_grid[:,:,k]*dhs[k] 
+        v_mean += v_grid[:,:,k]*dhs[k]
+        d_mean += div_grid[:,:,k]*dhs[k]
+    end
+
+    #Now use the mean fields
+    grad!(pres_surf[:,:,l2], dumc[:,:,2], dumc[:,:,3])
+    px = gridded(dumc[:,:,2]*3600, scale=true)
+    py = gridded(dumc[:,:,3]*3600, scale=true) #3600 factor from Paxton/Chantry. I think this is to correct for the underflow rescaling earlier
+
+    pres_surf_tend = spectral(-u_mean.*px - v_mean.*py)
+    pres_surf_tend[1,1] = Complex{NF}(zero)
+
+end
+
+"""
+Compute the spectral tendency of the "vertical" velocity
+"""
+function vertical_velocity_tendency!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                     Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                     C::Constants{NF}
+                                     ) where {NF<:AbstractFloat}
+
+    @unpack u_grid,v_grid,div_grid = Diag.gridvars
+    @unpack u_mean,v_mean,d_mean,sigma_tend,sigma_m,puv,px,py = Diag.miscvars
+    @unpack nlev,dhs = C 
+
+    #Initialise
+    sigma_tend[:,:,:] = 0.0
+    sigma_m[:,:,:] = 0.0
+
+
+    for k in 1:nlev
+        puv[:,:,k] = (u_grid[:,:,k] - u_mean)*px + (v_grid[:,:,k] - v_mean)*py
+    end
+
+    for k in 1:nlev
+        sigma_tend[:,:,k+1] = sigma_tend[:,:,k] - dhs[k]*(puv[:,:,k] + div_grid[:,:,k] - d_mean)
+        sigma_m[:,:,k+1] = sigma_m[:,:,k] - dhs[k]*puv[:,:,k]
+    end
+
+end
+
+
+
+"""
+Compute the temperature anomaly in grid point space
+"""
+function temperature_grid_anomaly!(Diag::PrognosticVariables{NF}, # Diagnostic variables
+                                   C::Constants{NF}
+                                   ) where {NF<:AbstractFloat}
+
+    @unpack temp_grid,temp_grid_anomaly = Diag.gridvars
+    @unpack nlev,tref = C  #Is tref a constant?
```

Comment:
There are currently two temperature reference values declared in `Params`, `temp_ref` the surface absolute temperature referece (288K) and `temp_top` the reference temperature of the stratosphere (216K), if you need more I'd add it to parameters.jl and convert it to the respective number format NF to be held in `Constants`

Reply:
In previous (fortran) Speedy we have `tref` and permutations defined as:

```fortran

do k=1,kx
        tref(k)=288.0_dp*max(0.2_dp,fsg(k))**rgam 
        tref1(k)=rgas*tref(k)
        tref2(k)=akap*tref(k)
        tref3(k)=fsgr(k)*tref(k)
end do
``` 

Like `dhs` I think these can ultimately be calculated in the Constants struct, right?